### PR TITLE
[Ready for Review] Add msgs serialization tests

### PR DESCRIPTION
**Pull request by ariard**

### CODE REVIEW

**Comment by ariard** [comment]
WIP, more in the way of already there encoding_announcement_signatures

**Comment by ariard** [comment]
Rebased

**Comment by ariard** [comment]
Good, tests for the 23 types of messages (bolt7 query-thing messages except)

File: src/ln/msgs.rs [hunk 1]
```diff
@@ -1429,4 +1441,612 @@ mod tests {
...
+		let sig_1 = get_sig_on!(privkey, secp_ctx, String::from("01010101010101010101010101010101"));
+		let sig_2 = get_sig_on!(privkey, secp_ctx, String::from("02020202020202020202020202020202"));
+		let announcement_signatures = msgs::AnnouncementSignatures {
+			channel_id: [4, 0, 0, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0],
+			short_channel_id: 2316138423780173,
+			node_signature: sig_1,
+			bitcoin_signature: sig_2,
+		};
+
+		let encoded_value = announcement_signatures.encode();
```
Comment (TheBlueMatt) [inline src/ln/msgs.rs]: nit: trailing whitespace.

File: src/ln/msgs.rs [hunk 2]
```diff
@@ -1429,4 +1436,251 @@ mod tests {
...
+		}
+		target_value.append(&mut hex::decode("00083a840000034d031b84c5567b126440995d3ed5aaba0565d71e1834604819ff9c17f5e9d5dd078f024d4b6cd1361032ca9bd2aeb9d900aa4d45d9ead80ac9423374c451a7254d076602531fe6068134503d2723133227c867ac8fa6c83c537e9a44c3c5bdbdcb1fe33703462779ad4aad39514614751a71085f2f10e1c7a593e4e030efb5b8721ce55b0b").unwrap());
+		if excess_data {
+			target_value.append(&mut hex::decode("0a00001400001e000028").unwrap());
+		}
+		assert_eq!(encoded_value, target_value);
+	}
+
+	#[test]
+	fn encoding_channel_announcement() {
```
[the remark this answers is not in the document]
Comment (ariard) [reply src/ln/msgs.rs]: Surely forget, fulfilled

File: src/ln/msgs.rs [hunk 3]
```diff
@@ -1683,4 +1688,297 @@ mod tests {
...
+		let (_, pubkey_2, _) = get_keys_from!("0202020202020202020202020202020202020202020202020202020202020202");
+		let (_, pubkey_3, _) = get_keys_from!("0303030303030303030303030303030303030303030303030303030303030303");
+		let (_, pubkey_4, _) = get_keys_from!("0404040404040404040404040404040404040404040404040404040404040404");
+		let (_, pubkey_5, _) = get_keys_from!("0505050505050505050505050505050505050505050505050505050505050505");
+		let (_, pubkey_6, _) = get_keys_from!("0606060606060606060606060606060606060606060606060606060606060606");
+		let open_channel = msgs::OpenChannel {
+			chain_hash: if !non_bitcoin_chain_hash { Sha256dHash::from_hex("6fe28c0ab6f1b372c1a6a246ae63f74f931e8365e15a089c68d6190000000000").unwrap() } else { Sha256dHash::from_hex("000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943").unwrap() },
+			temporary_channel_id: [2; 32],
+			funding_satoshis: 2316138423780173,
+			push_msat: 2316138423780173,
```
[the remark this answers is not in the document]
Comment (ariard) [reply src/ln/msgs.rs]: Yes, generated others test vectors for accept_channel/open_channel and update_add_htlc

File: src/ln/msgs.rs [hunk 4]
```diff
@@ -1683,4 +1688,297 @@ mod tests {
...
+		assert_eq!(encoded_value, target_value);
+	}
+
+	fn do_encoding_shutdown(script_type: u8) {
+		let (_, pubkey_1, _) = get_keys_from!("0101010101010101010101010101010101010101010101010101010101010101");
+		let script = Builder::new().push_opcode(opcodes::OP_TRUE).into_script();
+		let shutdown = msgs::Shutdown {
+			channel_id: [2; 32],
+			scriptpubkey: if script_type == 1 { Address::p2pkh(&pubkey_1, Network::Testnet).script_pubkey() } else if script_type == 2 { Address::p2sh(&script, Network::Testnet).script_pubkey() } else if script_type == 3 { Address::p2wpkh(&pubkey_1, Network::Testnet).script_pubkey() } else { Address::p2wsh(&script, Network::Testnet).script_pubkey() },
+	 	};
```
[the remark this answers is not in the document]
Comment (ariard) [reply src/ln/msgs.rs]: K

File: src/ln/msgs.rs [hunk 5]
```diff
@@ -1429,4 +1436,251 @@ mod tests {
 			vec![4, 0, 0, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 0, 6, 0, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 3, 0, 0, 0, 0, 0, 0, 0, 4, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 3, 27, 132, 197, 86, 123, 18, 100, 64, 153, 93, 62, 213, 170, 186, 5, 101, 215, 30, 24, 52, 96, 72, 25, 255, 156, 23, 245, 233, 213, 221, 7, 143]
 		);
 	}
+
+	macro_rules! get_keys_from {
```
[the remark this answers is not in the document]
Comment (ariard) [reply src/ln/msgs.rs]: Okay, shared secp_context between keys belonging to same test

File: src/ln/msgs.rs (hunk 1 above)
Comment (ariard) [reply src/ln/msgs.rs]: K

**Comment by ariard** [comment]
Hmm seems fail is due to travis issue, thread exists forcefully and first time I see these warnings : "Your glibc version:'2.19' will most likely result in malloc()-related deadlocks. Min. version 2.24 (Or, Ubuntu's 2.23-0ubuntu6) suggested. See https://sourceware.org/bugzilla/show_bug.cgi?id=19431 for explanation. Using clone() instead of fork()" ?

**Pull request merged**